### ISSUE TITLE
Turn noise level in GaussianFitter into instance variables

### ISSUE THREAD

**Issue by SpencerFleming** [BoiseState-AdaptLab/adapt-lidar-tools#298]
In GaussianFitter::guess_peaks:
```
625     //UPDATE: We are only using a noise level of 6 because we want all peaks
626     //with an amplitude >= 10
627     //Level up to and including which peaks will be excluded
628     //For the unaltered wave, noise_level = 16
629     //for the second derivative of the wave, noise_level = 3
630     // this is creating guesses for a guassian fitter that does not do
631     // well if we have guesses that have an amplitude more than an order
632     // of magnitute apart. We are going to set the noise level to be the
633     // max value/ 10 - max*.05;
634     max = 0;
635     for(int i = 0; i<(int)ampData.size(); i++){
636         if(ampData[i]>max){
637             max = ampData[i];
638         }
639     }
640     noise_level = ((float)max)*.09;
```
```
647     if (noise_level < 6){
648         noise_level = 6;
649     }
```
Like several other magic numbers, this involves some variable-dependent expressions to determine it. I'll have to ask about it.

**Comment by SpencerFleming** [comment]
I have worked on all the other magic numbers and was able to design simple solutions to accommodate their dependencies (generally I found it was okay to just take the magic part of the expressions, turn them into instance variables, and comment on the side how they are modified before use.

This case is different because I know it's been experimented with, and I'm not sure how much of all the relevant code is something we were planning on keeping, or patchwork done for testing purposes. I'll ask about it.
